test/StreamJsonRpc.Tests/MarshalableProxyTests.cs

```diff
@@ -0,0 +1,522 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using MessagePack;
+using Microsoft;
+using Microsoft.VisualStudio.Threading;
+using Nerdbank.Streams;
+using Newtonsoft.Json;
+using StreamJsonRpc;
+using Xunit;
+using Xunit.Abstractions;
+
+/// <summary>
+/// Tests the proxying of interfaces marked with <see cref="RpcMarshalableAttribute"/>.
+/// </summary>
+public abstract class MarshalableProxyTests : TestBase
+{
+    protected readonly Server server = new Server();
+    protected readonly JsonRpc serverRpc;
+    protected readonly JsonRpc clientRpc;
+    protected readonly IServer client;
+
+    protected MarshalableProxyTests(ITestOutputHelper logger)
+        : base(logger)
+    {
+        var pipes = FullDuplexStream.CreatePipePair();
+
+        this.client = JsonRpc.Attach<IServer>(new LengthHeaderMessageHandler(pipes.Item1, this.CreateFormatter()));
+        this.clientRpc = ((IJsonRpcClientProxy)this.client).JsonRpc;
+
+        this.serverRpc = new JsonRpc(new LengthHeaderMessageHandler(pipes.Item2, this.CreateFormatter()));
+        this.serverRpc.AddLocalRpcTarget(this.server);
+
+        this.serverRpc.TraceSource = new TraceSource("Server", SourceLevels.Verbose);
+        this.clientRpc.TraceSource = new TraceSource("Client", SourceLevels.Verbose);
+
+        this.serverRpc.TraceSource.Listeners.Add(new XunitTraceListener(this.Logger));
+        this.clientRpc.TraceSource.Listeners.Add(new XunitTraceListener(this.Logger));
+
+        this.serverRpc.StartListening();
+    }
+
+    [RpcMarshalable]
+    public interface IMarshalable : IDisposable
+    {
+        Task DoSomethingAsync();
+    }
+
+    [RpcMarshalable]
+    public interface IGenericMarshalable<T> : IMarshalable
```

Comment:
How about a negative test for a non-attributed interface that derives from an attributed interface?